modules/videoio/perf/perf_input.cpp

```diff
@@ -32,4 +33,71 @@ PERF_TEST_P(VideoCapture_Reading, ReadFile, testing::ValuesIn(bunny_files) )
   SANITY_CHECK_NOTHING();
 }
 
+PERF_TEST(VideoCapture_Reading_1000_frame, Get1000frame)
+{
+    VideoCapture cap1(0);
+    VideoCapture cap2(2);
+
+    ASSERT_TRUE(cap1.isOpened());
+    ASSERT_TRUE(cap2.isOpened());
+
+    Mat frame1;
+    int ITERATION_COUNT = 500;
+
+    //false start
+    cap1>>frame1;
+    cap2>>frame1;
+
+    TEST_CYCLE() {
+        for(int j = 0; j < ITERATION_COUNT; ++j)
+        {
+            cap1>>frame1;
+            cap2>>frame1;
+        }
+    };
+    SANITY_CHECK_NOTHING();
+}
+
+PERF_TEST(VideoCapture_Reading_1000_frame, GetWaitAny1000frame)
+{
+    VideoCapture cap1(0);
+    VideoCapture cap2(2);
+
+    ASSERT_TRUE(cap1.isOpened());
+    ASSERT_TRUE(cap2.isOpened());
+
+    std::vector<VideoCapture> VCM;
+
+    VCM.push_back(cap1);
+    VCM.push_back(cap2);
+
+    std::vector<int> state;
+
+    Mat frame1, frame2;
+    std::vector<Mat> forMAt = {frame1, frame2};
+
+    int ITERATION_COUNT = 500;
+    int TIMEOUT = -1;
+
+    //false start
+    cap1>>frame1;
+    cap2>>frame2;
+
+    TEST_CYCLE() {
+        int out = 0;
+        for(int j = 0; j < ITERATION_COUNT; ++j)
+        {
+            VideoCapture::waitAny(VCM, state, TIMEOUT);
+            for(int i = 0; i < 2; ++i)
+            {
+                if(state[i] == CAP_CAM_READY)
+                {
+                  EXPECT_TRUE(VCM[i].retrieve(forMAt[i]));
```

Comment:
Perhaps performance tests do not make much sense in this case. Standalone benchmark application (in `samples/cpp` dir) would be more useful. Lets say it could be launched in two modes: sync and async, and will perform some processing on frames (e.g. Canny) and measure performance and efficiency metrics. Take a look at the [videocapture_camera.cpp](https://github.com/opencv/opencv/blob/master/samples/cpp/videocapture_camera.cpp) sample application, it can be used as a starting point.